### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by iamsjunaid**
Title: PR Summary - Adding Current User Functionality, Post and Comment Creation, and Likes

Description:

This pull request introduces several new features and enhancements to the application, addressing various functionalities and user interactions. The primary focus is on enhancing user experience and interaction within the application.

**1. Introducing current_user Method in ApplicationController:**

A new method named `current_user` has been added to the `ApplicationController`. This method ensures that user-specific data is accessible across all controllers. The `current_user` method allows for seamless integration of user-specific actions throughout the application.

**2. Post and Comment Creation:**

Forms have been implemented to enable users to create posts and comments. These forms are designed to leverage the `current_user` method previously introduced in the `ApplicationController`. This ensures that posts and comments are associated with the correct user who created them.

- Users can now create new posts through a user-friendly form. The creation process utilizes the `current_user` method, making it easier to attribute posts to the relevant user.

- The commenting functionality has also been implemented. Users can now leave comments on posts, and the comments are linked to the user who wrote them, thanks to the `current_user` method.

**3. Adding Likes to Posts:**

A significant enhancement to user engagement is the addition of a "like" feature for posts. Users now have the ability to express appreciation for posts they enjoy. This feature enhances user interaction and encourages positive engagement within the application.

These changes collectively contribute to a more interactive and engaging user experience. By making user-specific data accessible, enabling post and comment creation, and allowing users to express appreciation through likes, this pull request significantly enhances the overall application functionality and user satisfaction.

The PR is ready for review, and feedback is welcome to ensure the successful integration of these features into the application.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -8,4 +8,29 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    puts 'Current User'
```
Comment (TedLivist) [inline app/controllers/posts_controller.rb]: - [x] Kindly remove this print statement. It is generally not a good practice to print statements. Kindly cleanup your code.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  def create
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
+    @comment.post = Post.find(params[:post_id])
+
+    respond_to do |_format|
+      if @comment.save
+
+        redirect_to user_post_path(user_id: params[:user_id], id: params[:post_id])
```
Comment (TedLivist) [inline app/controllers/comments_controller.rb:14]: - [ ] **[OPTIONAL]** From your code a comment is created users should be redirected to the post, but that does not happen, the app remains on the create new comment page. You may want to look into that.

  image

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -8,4 +8,29 @@ def show
...
+
+    @post = Post.new(post_params)
+    @post.author = current_user
+
+    respond_to do |_format|
+      if @post.save
+        redirect_to user_posts_path(current_user)
```
Comment (TedLivist) [inline app/controllers/posts_controller.rb:22]: - [ ] **[OPTIONAL]** From your code a post is created users should be redirected to the post, but that does not happen, the app remains on the new post page. You may want to look into that.

  image

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @iamsjunaid,

Good job so far!
✔️ Posts, comments and likes can be created from the UI
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -8,4 +8,27 @@ def show
...
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+
+    respond_to do |_format|
+      if @post.save
```
Comment (aamirkhan2478) [inline app/controllers/posts_controller.rb:21]: **[OPTIONAL but highly recommended]** As mentioned by the previous :point_up: reviewer, in your code a post is created users should be redirected to the post, but that does not happen, the app remains on the new post page. You may want to look into that so kindly implement this as well :slightly_smiling_face:

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (aamirkhan2478) [inline app/controllers/comments_controller.rb:14]: **[OPTIONAL but highly recommended]** As mentioned by the previous ☝️ reviewer,  in your code a comment is created users should be redirected to the post, but that does not happen, the app remains on the create new comment page. You may want to look into that so kindly fix this issue as well :slightly_smiling_face:

**Review by aamirkhan2478 — APPROVED** [review]
Hi @iamsjunaid, 

GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ You used Gitflow correctly :+1: 
✅ No linter error :+1: 
✅ Good commit messages :+1: 
✅ Good descriptive PR :+1: 
✅ Well done :clap: you did all changes given by the previous reviewer :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**